pkg/controllers/ippool/ippool_controller.go

```diff
@@ -47,24 +48,29 @@ type IPPoolReconciler struct {
 	Scheme     *apimachineryruntime.Scheme
 	Service    *ippool.IPPoolService
 	VPCService servicecommon.VPCServiceProvider
+	Recorder   record.EventRecorder
 }
 
-func deleteSuccess(r *IPPoolReconciler, _ *context.Context, _ *v1alpha2.IPPool) {
+func deleteSuccess(r *IPPoolReconciler, _ *context.Context, o *v1alpha2.IPPool) {
+	r.Recorder.Event(o, v1.EventTypeNormal, "SuccessfulDelete", "SuccessfulDelete: IPPool CR has been successfully deleted")
```

Comment:
If CR is deleted successful, we do not need to create event as CR is deleted, right?

Reply:
I searched other pages saying that there are normal events below we could record:

ObjectCreated: Record when a new object is created.
ObjectUpdated: Record when an object is updated.
ObjectDeleted: Record when an object is deleted.
ObjectFinalized: Record when an object is finalized.
ObjectInitialized: Record when an object is initialized.

 Also Custom Events:

> You can also define custom events based on the specific logic of your controller. For example, if there are certain conditions that need to be highlighted during reconciliation, you can create custom events for those conditions.